local/local.go

```diff
@@ -87,13 +102,98 @@ func NewImage(repoName string, dockerClient client.CommonAPIClient, ops ...Image
 		downloadBaseOnce: &sync.Once{},
 	}
 
-	for _, v := range ops {
-		image, err = v(image)
-		if err != nil {
-			return nil, err
-		}
+	image, err = processPreviousImageOption(image, imageOpts.prevImageRepoName, platform, dockerClient)
+	if err != nil {
+		return image, err
```

Comment:
```suggestion
		return nil, err
```
We typically return empty values for all other return values, when error in non-nil

Reply:
Now fixed in local and remote